CONTRIBUTING.md

```diff
@@ -5,82 +5,182 @@ may be helpful to understand the goal of the project. The goal of CometBFT is to
 develop a BFT consensus engine robust enough to support permissionless
 value-carrying networks. While all contributions are welcome, contributors
 should bear this goal in mind in deciding if they should target the main
-CometBFT project or a potential fork. When targeting the main CometBFT project,
-the following process leads to the best chance of landing changes in `main`.
-
-All work on the code base should be motivated by a [GitHub
-Issue](https://github.com/cometbft/cometbft/issues).
-[Search](https://github.com/cometbft/cometbft/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
-is a good place to start when looking for places to contribute. If you would
-like to work on an issue which already exists, please indicate so by leaving a
-comment.
-
-All new contributions should start with a [GitHub
-Issue](https://github.com/cometbft/cometbft/issues/new/choose). The issue helps
-capture the problem you're trying to solve and allows for early feedback. Once
-the issue is created the process can proceed in different directions depending
-on how well defined the problem and potential solution are. If the change is
-simple and well understood, maintainers will indicate their support with a
-heartfelt emoji.
+CometBFT project or a potential fork.
+
+## Overview
+
+When targeting the main CometBFT project, following the processes outlined in
+this document will lead to the best chance of landing changes in a release.
+
+### Core team responsibility
+
+First and foremost, the CometBFT core team is responsible for maintaining this
+project over time. This means that the core team needs to understand the nature
+of, and agree to maintain, all of the changes that land on `main` or a backport
+branch. It may cost you a few days/weeks' worth of your time to submit a
+particular change, but it may cost the core team far more to maintain that
+change over the years.
```

Comment:
We could rephrase this:

```suggestion
branch. It may cost a few days/weeks' worth of time to _submit_ a
particular change, but _maintaining_ that change over the years has a
much higher cost that the core team will bear.
```